test-data-generator/src/main/scala/za/co/absa/spline/testdatagen/GenCLI.scala

```diff
@@ -101,9 +105,14 @@ object GenCLI {
 
   private def createDispatcher(name: String, config: Config): FileDispatcher = name match {
     case "file" =>
-      new FileDispatcher(s"${config.graphType.name}-lineage-" +
-        s"${config.reads}reads-" +
-        s"${config.operations}ops-" +
-        s"${config.attributes}attr")
+      config.customOutputFileName.fold {
+        val defaultPrefixName = s"${config.graphType.name}-lineage-" +
+          s"${config.reads}reads-" +
+          s"${config.operations}ops-" +
+          s"${config.attributes}attr"
+        new FileDispatcher(defaultPrefixName) // using default prefix and default built-in suffix
+      } { customFileName =>
+        new FileDispatcher(customFileName, "") // custom name as-is
```

Comment:
the filename creation seems to be somehow spread over two classes. I think better would be to just have a method for creating the default filename (including the suffix).
```Scala
new FileDispatcher(createDefaultFileName(config))
```

Then you don't need `fileNameSuffix` in FileDispatcher

Reply:
👍 Good idea, done.